### ISSUE TITLE
Detect outdated node.js version

### ISSUE THREAD

**Issue by MartinKolarik** [jsdelivr/globalping-probe#171]
- Option 1 - during the auto-update and don't apply the update (can break API communication later)
 - Option 2 - during app startup (after the update) - only log an error and exit
 - ???

Probably option 2 but we can decide after #168.

**Comment by MartinKolarik** [comment]
Option two seems reasonable. In the future, for probes connected to the dashboard, we can also email their users if they run on an outdated version.

**Comment by MartinKolarik** [comment]
Right now, it's just about adding the mechanism. We can still keep supporting node 16. But when it goes out of life (soon), we won't support it anymore, and then it's better to ask clearly for an update than have it crash with a random error due to some missing API.

**Comment by jimaek** [comment]
Ok that part wasn't clear from the description

**Comment by MartinKolarik** [comment]
Indeed, the task is to check if the installed version matches what's defined in `engines` - if no, alert the user and exit. We have 16 there right now, so it won't have an immediate impact. But 16 reaches EOL on September 11th, 2023, so after that, it'll need to be removed, and then the probes will stop working.

Maybe we can also add a warning mechanism: right now, if the check passes, but 16 is detected, print a message that it won't be supported soon with instructions on how to upgrade. In the future, we can always activate the warning two months in advance or so. The LTS versions of node have a three years life time, so we'll have this issue again in April 2025.

**Comment by MartinKolarik** [comment]
For extra context: we can't keep supporting EOL versions not only because it would limit us directly but most package maintainers drop support for EOL node versions quickly, so it would prevent us from updating any of the dependencies.

**Comment by jimaek** [comment]
Sounds good, but we dont have to hurry if there is no need for it. We can continue supporting 16 for much longer as long as it doesn't start creating issues for us. Freezing package versions is also an option.

Its in our best interest to keep as many probes online as possible. And since most people wont be reading the logs they wont even notice that it broke.

Killing version support can only be discussed after the dashboard goes live and we finally have a method of reaching the users. 

We should also add a mechanism to advertise this, something like "Adopt this probe to generate free credits, manage your probes and get notified when something happens. Visit globalping.io/adopt"

**Comment by MartinKolarik** [comment]
The dashboard isn't so relevant here. It's nice to have, but it requires the user to manually register the probe, which means he needs to interact with it, and if he does that, he will notice the log error too. Someone who sets it up and forgets about it won't connect to the dashboard, either.

**Comment by jimaek** [comment]
There is a chance that person follows us somewhere and will be motivated to check out his probe if he sees the benefit of adoption. In any case I dont want to kill any versions regardless of timelines until its absolutely necessary. And the sooner we have the dashboard the sooner we get a tool to notify the users.

**Comment by MartinKolarik** [comment]
This has been partly resolved by https://github.com/jsdelivr/globalping-probe/issues/192, but because auto-update doesn't apply to probes with < 1 GB of RAM, we should do some additional things mentioned here:
 - log a big warning [here](https://github.com/jsdelivr/globalping-probe/blob/9f485606bb95dad273cc01539affbc3c9cc16b0a/src/index.ts#L42) if the version is outdated and we can't update it,
 - propagate the node version to the dashboard so that we can show warnings for adopted probes.
